lib/async/redis/sentinels.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module Async
+	module Redis
+		class SentinelsClient < Client
+			def initialize(master_name, sentinels, role = :master, protocol = Protocol::RESP2, **options)
+				@master_name = master_name
+				@sentinel_endpoints = sentinels.map do |sentinel|
+					Async::IO::Endpoint.tcp(sentinel[:host], sentinel[:port])
+				end
+				@role = role
+
+				@protocol = protocol
+				@pool = connect(**options)
+			end
+
+			private
+
+			# Override the parent method. The only difference is that this one needs
+			# to resolve the master/slave address.
+			def connect(**options)
+				Async::Pool::Controller.wrap(**options) do
+					address = resolve_address
+					endpoint = Async::IO::Endpoint.tcp(address[:ip], address[:port])
+
+					peer = endpoint.connect
+					stream = IO::Stream.new(peer)
+
+					@protocol.client(stream)
+				end
+			end
+
+			def resolve_address
+				address = case @role
+				when :master
+					resolve_master
```

Comment:
I think this is simpler if you write `return resolve_master` and the final statement is `raise RuntimeError, ...`

Reply:
Not sure I follow. Both `resolve_master` and `resolve_slave` can return nil. So we can't return without checking nil unless we move `raise RuntimeError ... ` to the `Async::Pool::Controller` block above.

Reply:
```ruby
def foo
  return nil
end

x, y = foo
x # nil
y # nil
```

Reply:
Oh, I see, the nil check... hmm, it just seems like something that could be a bit simpler. But if not that's okay.

Reply:
I guess the question I would ask, is there any situation where returning nil from `resolve_*` is useful?

Or is it always triggering an exception?

User may do `resolve_master || resolve_slave`. Then, returning `nil` makes total sense.

Yes, it could make sense to throw the exception in `resolve_master`. At that point, you also have more information available to give a meaningful error message.

Reply:
I don't think the user should be able to do `resolve_master || resolve_slave`. I think the user needs to decide whether the requests should go to a master or slave when instantiating the client, but allowing both at the same time is a bit dangerous.

Some users might know, for example, that for their use case they only need to perform read operations and they don't mind about the master->slave replication lag. I that case, they might decide to perform those reads on the slaves to decrease the load of the master.